internal/scheduler/scheduler.go

```diff
@@ -273,13 +285,39 @@ func (s *Scheduler) syncState(ctx context.Context) ([]*jobdb.Job, error) {
 		jobsToUpdateById[jobId] = job
 	}
 
+	// Ensure queued jobs being retried have node anti affinities for any nodes they have already run on
+	for _, job := range jobsToUpdateById {
```

Comment:
can this be simplified?  We seem to iterate over `jobsToUpdateById` twice and do much the same thing in each case?

It feels like we could iterate over once and apply the anti affinity in one place.